distsql/select_result.go

```diff
@@ -114,9 +114,9 @@ func (r *selectResult) NextRaw(ctx context.Context) ([]byte, error) {
 }
 
 // Next reads data to the chunk.
-func (r *selectResult) Next(ctx context.Context, chk *chunk.Chunk) error {
-	chk.Reset()
-	for chk.NumRows() < r.ctx.GetSessionVars().MaxChunkSize {
+func (r *selectResult) Next(ctx context.Context, chk *chunk.Chunk, maxChunkSize int) error {
+	chk.GrowAndReset(maxChunkSize)
+	for chk.NumRows() < chk.Capacity() {
```

Comment:
It should be `maxChunkSize`? Because `chk.Capacity()` may be larger than `maxChunkSize`.

Reply:
`chk.Capacity()` never grow over than `maxChunkSize`, but `NumRow` maybe larger than maxChunkSize; 

```
capacity < maxChunkSize
0 < numRow (almost < capacity but sometime > capacity)
```
chunk's naming often misleading :rofl:

Reply:
Use `maxChunkSize` improves the code readability.

Reply:
@zz-jason I remember this name is zhou sir's advise in a meeting room, so let's rename it for this chance? :rofl:

Reply:
No, I mean this:
```go
for chk.NumRows() < maxChunkSize {
```

Not changing the function name from `Capacity()` to `maxChunkSize()`.